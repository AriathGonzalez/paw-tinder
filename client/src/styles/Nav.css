nav {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 100%;
}

nav .logo-container {
  width: 15%;
  margin: 1em;
}

nav .logo-container .logo {
  width: 100%;
}

nav .nav-button {
  color: rgba(244, 65, 108, 1);
  cursor: pointer;
  background-color: rgba(249, 249, 249, 1);
  font-size: 1.25rem;
  padding: 0.65em 1.25em;
  border-radius: 0.65em;
  border: none;
  margin: 0.65em;
}

nav .nav-button:disabled {
  background-color: rgba(179, 170, 170, 0.6);
  cursor: default;
}
