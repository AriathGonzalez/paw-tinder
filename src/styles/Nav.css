nav {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 100%;
}

nav .logo-container {
  width: 170px;
  margin: 15px;
}

nav .logo-container .logo {
  width: 100%;
}

nav .nav-button {
  color: rgba(244, 65, 108, 1);
  background-color: white;
  font-size: 20px;
  padding: 10px 20px;
  border-radius: 10px;
  border: none;
  margin: 10px;
}
