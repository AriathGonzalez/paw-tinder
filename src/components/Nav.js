import colorLogo from "../images/paw_tinder_logo.png";
import blackLogo from "../images/paw_tinder_logo_black.png";
import "../styles/Nav.css";

const Nav = ({ minimal, authToken, showModal, setShowModal, setIsSignUp }) => {
  const handleClick = () => {
    setShowModal(true);
    setIsSignUp(false);
  };

  return (
    <nav>
      <div className="logo-container">
        <img
          className="logo"
          src={minimal ? colorLogo : blackLogo}
          alt="Logo"
        />
      </div>

      {!authToken && !minimal && (
        <button
          className="nav-button"
          onClick={handleClick}
          disabled={showModal}
        >
          Log In
        </button>
      )}
    </nav>
  );
};

export default Nav;
