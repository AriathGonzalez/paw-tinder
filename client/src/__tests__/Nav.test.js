import Nav from "../components/Nav";
import renderer from "react-test-renderer";

describe("Nav Component", () => {
  describe("Snapshot Test", () => {
    it("Should render correctly", () => {
      const tree = renderer
        .create(
          <Nav
            minimal={false}
            showModal={false}
            setShowModal={null}
            setIsSignUp={null}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
