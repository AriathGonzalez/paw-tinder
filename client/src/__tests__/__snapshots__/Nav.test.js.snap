// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component Snapshot Test Should render correctly 1`] = `
<nav>
  <div
    className="logo-container"
  >
    <img
      alt="Logo"
      className="logo"
      src="paw_tinder_logo_black.png"
    />
  </div>
</nav>
`;
